PR-Review-Guide.md

```diff
@@ -0,0 +1,139 @@
+
+### Design
+
+- Could the code be simplified / refactored?
+  - Is the code "too clever"? Overly clever code tends to be fragile: reject it unless there is a **very** good documented reason!
+- Is there a better (simpler) solution to the problem?
+- Is there any duplication?
+- Is the code "re-inventing the wheel?" Is there a standard package that could be used instead?
+- What assumptions does the code make? Are they valid? Are they documented?
+- Does the code make sense?
+  - Could someone with minimal exposure to the codebase guess at what it is doing? (hint: they should be able to).
+
+### Code comments
+
+- Is the code sufficiently well commented?
+- Are all functions and function parameters documented (or for `golang` programs, is their purpose obvious?)
+- Does this change require an update to the `README`(s) / architecture docs?
+- Are any lines commented out? If so, they should be **removed**!
+- Are there any `FIXME`/`TODO`'s? If so, they should be replaced by a full URL to an issue number tracking the work that needs to be done.
+
+### Documentation
+
+- Does the document update need to be applied to any **other** documents?
+  For example, when a PR updates an installation guide, those same changes may need to be applied to one or more of the other installation guides!
+
+- Does the PR make any of the following changes? If so, it may well need a documentation change:
+  - new programs / scripts added or removed.
+  - new program / script options added / removed.
+  - new config options added or removed.
+  - new features added / removed.
+  - existing bug(s) fixed.
+
+### Logging and Debugging
+
+- Could this code be debugged from a logfile if it failed?
+- Are all fatal scenarios logged?
+- Are error messages and log calls useful and comprehensive? Ensure sensitive information is **not** displayed though.
+
+### Testing
+
+- Has the code been written to be easy to test?
+- Are there new unit tests? If not, why not?
+- What other classes of testing (such as integration, system, stress, fuzz) could be written to check the code?
+
+### Environments
+
+- Will the code work on all distributions?
+- Will the code run on different architectures?
+- What environments will this code run in?
+- Which user will the code run as? If it needs to run as the superuser, it should have clear documentation explaining why this is necessary.
+
+### Upgrades
+
+- If the code logs internal state to a file or shared memory, how will upgrades be handled?
+  For example, if a new version introduces a new state file format, can the
+  new version consume the old state file format files?
+- If upgrades cannot be handled, this should be documented.
+
+## Mandatory Checks
+
+- Provide constructive comments on the code.
+  - If you like some aspect particularly, say so! :smile:
+  - If you don't like something, state that too, but respectfully!
+  - If you don't understand the code, don't be afraid to say so - code should
```

Comment:
- If you do not understand the code, say so because code should **always** be clear.